crates/polars-expr/src/expressions/mod.rs

```diff
@@ -538,6 +538,20 @@ pub trait PhysicalExpr: Send + Sync {
     /// Take a DataFrame and evaluate the expression.
     fn evaluate(&self, df: &DataFrame, _state: &ExecutionState) -> PolarsResult<Column>;
 
+    /// Attempt to cheaply evaluate this expression in-line without a DataFrame context.
+    /// This is used by StatsEvaluator when skipping files / row groups using a predicate.
+    /// TODO: Maybe in the future we can do this evaluation in-line at the optimizer stage?
+    ///
+    /// Do not implement this directly - instead implement `evaluate_inline_impl`
+    fn evaluate_inline(&self) -> Option<Column> {
```

Comment:
Could we achieve the same by passing an empty `DataFrame` and if it raises change the error to `Option`?

Reply:
I think we could, but we would lose a bit of control - I'm currently limiting the evaluation depth as well as in some places (e.g. `Cast`), I only evaluate the expression if the input length is 1, to avoid evaluating large expressions.